### PR TITLE
Added foreign keys for cms_users

### PR DESCRIPTION
**Pull request by naphets123**

### CODE REVIEW

**Comment by naphets123** [comment]
These are the migrations for history transactions, and log as there is still open question about adding foreign keys to "modified by" and "created by"

**Comment by jamescrowley** [comment]
> These are the migrations for history transactions, and log as there is still open question about adding foreign keys to "modified by" and "created by"

which table has the modified by and created by on them?

File: db/migrations/20190923101044_foreign_keys_users.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    ;
+
+        $this->table('transactions')
+            ->addForeignKey('user_id', 'cms_users', 'id', [
+                'delete' => 'RESTRICT', 'update' => 'CASCADE',
+            ])
+            ->save()
+    ;
+        $this->table('log')
+            ->changeColumn('user', 'integer', [
```
[the remark this answers is not in the document]
Comment (HaGuesto) [reply db/migrations/20190923101044_foreign_keys_users.php]: annoying. Sorry, I did not check this properly.
mmh. Do we actually show the `log` table anywhere? It actually just tracks when somebody logs in and out.

I guess we can neglect that one.

**Comment by HaGuesto** [comment]
> > These are the migrations for history transactions, and log as there is still open question about adding foreign keys to "modified by" and "created by"
> 
> which table has the modified by and created by on them?

Any menu which you can edit through the UI. Check out this function
https://github.com/boxwise/dropapp/blob/0fd2aa6b22866c5ff9aa2fdf0e62d27d03e6a480/library/lib/formhandler.php#L216

**Comment by jamescrowley** [comment]
@naphets just a note that given the file naming I'm going to have to merge your PRs in sequence, unless we update each as we go.

**Pull request merged**